src/IncrementalFetchingHelper.php

```diff
@@ -105,7 +148,56 @@ public function processItem(object $item, int $rowIndex): void
             ));
         }
 
-        $this->maxValue = max($this->maxValue, $newValue);
+        // Check if is type allowed, for the first row
+        if (!$this->valueTypeChecked) {
+            if (!in_array($newType, self::ALLOWED_TYPES, true)) {
+                throw new UserException(sprintf(
+                    'Unexpected type "%s" of the incremental fetching "%s" key. Allowed types "%s".',
+                    $newType,
+                    $this->key,
+                    implode('", "', self::ALLOWED_TYPES)
+                ));
+            }
+
+            if ($newType === EdmType::STRING) {
```

Comment:
Tu varujeme pouzivatela, ze mu to spadne ak sa najde nejaka string hodnota, ktora ma inu dlzku ako ostatne (v tomto alebo dalsich spusteniach).